odl/discr/discr_ops.py

```diff
@@ -130,110 +144,212 @@ def finite_diff(f, out=None, axis=0, dx=1.0, edge_order=2,
 
     method, method_in = str(method).lower(), method
     if method not in ('central', 'forward', 'backward'):
-        raise ValueError("method '{}' has to be central, forward or backward"
+        raise ValueError('method {} is not central, forward or backward'
                          ''.format(method_in))
 
+    if isinstance(padding, (int, float)) or padding in (None, 'edge'):
+        if isinstance(padding, int):
+            padding = float(padding)
+    else:
+        raise ValueError('padding value ({}) not valid'.format(padding))
+
+    if edge_order is None:
```

Comment:
Holger is right. Per default no padding is used and then the edge order depends on the method of choice.